specs/HubProtocol.md

```diff
@@ -131,7 +131,17 @@ Keep alive behavior is achieved via the `Ping` message type. **Either endpoint**
 
 Ping messages do not have any payload, they are completely empty messages (aside from the encoding necessary to identify the message as a `Ping` message).
 
-It is up to the server implementation to decide how frequently (if at all) `Ping` frames are sent. The ASP.NET Core implementation sends `Ping` frames only when using the Server Sent Events and WebSockets transports, at a default interval of 15 seconds (configurable). However, a `Ping` frame is only sent if 15 seconds elapses since the last message was sent. Clients may choose to use the "Ping rate" to provide a timeout for the server connection. Since the Client can expect the server to send `Ping` frames at regular intervals, even when the connection is idle, it can use that to determine if the server has left without closing the connection. The ASP.NET Core implementation (both JavaScript and C#) use a default timeout window of 30 seconds, which is twice the server ping rate interval.
+The default ASP.NET implementation automatically pings both directions on active connections. These pings are at regular intervals, and allow detection of unpexpected disconnects (for example, unplugging a server). If the client detects that the server has stopped pinging, the client will close the connection, and vice versa.
+
+|  | Option name | Default value |
+|-|-|-|
+|**Server**: how often to ping client | ```HubOptions.KeepAliveInterval``` | 15 seconds |
+|**Server**: if the client hasn't messaged in this interval, close the connection | ```HubOptions.ClientTimeout``` | 30 seconds |
+|**Client**: how often to ping server | ```HubConnection.PingInterval``` | 15 seconds |
+|**Client**: if the server hasn't messsaged in this interval, close the connection | ```HubConnection.ServerTimeout``` | 30 seconds |
```

Comment:
Actually, I think we should avoid putting details about the options in the protocol spec. The goal of this document is just to specify the _protocol_ rather than the full behavior of the client and server. Also, this just becomes one more place to keep up-to-date as we change implementation logic :).

This would fit better on our documentation site. I'll file an issue there and our doc team can take a look.

For now, let's remove the table. The rest of this looks good though!

Reply:
Filed https://github.com/aspnet/Docs/issues/6599